kubernetes/gitlab.yml

```diff
@@ -0,0 +1,253 @@
+              name: gitlab-config
+              key: GITLAB_BACKUP_TIME
+
+        - name: DB_TYPE
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: DB_TYPE
+        - name: DB_HOST
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: DB_HOST
+        - name: DB_PORT
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: DB_PORT
+
+        - name: REDIS_HOST
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: REDIS_HOST
+        - name: REDIS_PORT
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: REDIS_PORT
+
+        - name: SMTP_ENABLED
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_ENABLED
+        - name: SMTP_DOMAIN
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_DOMAIN
+        - name: SMTP_HOST
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_HOST
+        - name: SMTP_PORT
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_PORT
+        - name: SMTP_USER
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_USER
+        - name: SMTP_STARTTLS
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_STARTTLS
+        - name: SMTP_AUTHENTICATION
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: SMTP_AUTHENTICATION
+
+        - name: IMAP_ENABLED
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_ENABLED
+        - name: IMAP_HOST
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_HOST
+        - name: IMAP_PORT
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_PORT
+        - name: IMAP_USER
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_USER
+        - name: IMAP_SSL
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_SSL
+        - name: IMAP_STARTTLS
+          valueFrom:
+            configMapKeyRef:
+              name: gitlab-config
+              key: IMAP_STARTTLS
+        ports:
+        - name: http
+          containerPort: 80
+        - name: ssh
+          containerPort: 22
+        volumeMounts:
+        - mountPath: /home/git/data
+          name: data
+        livenessProbe:
+          httpGet:
+            path: /
```

Comment:
This livenessProve doesn't work because "/" responds "404".
So this path should be "/-/liveness" and to set "GITLAB_MONITORING_IP_WHITELIST" is needed.